chemprop/featurizers/molecule.py

```diff
@@ -41,3 +43,28 @@ class MorganBinaryFeaturizer(MorganFeaturizerMixin, BinaryFeaturizerMixin, Vecto
 @MoleculeFeaturizerRegistry("morgan_count")
 class MorganCountFeaturizer(MorganFeaturizerMixin, CountFeaturizerMixin, VectorFeaturizer[Mol]):
     pass
+
+
+@MoleculeFeaturizerRegistry("rdkit_2d")
+class RDKit2DFeaturizer(VectorFeaturizer[Mol]):
+    def __init__(self):
+        warnings.warn(
+            "The RDKit 2D features can deviate signifcantly from a normal distribution. Consider "
+            "manually scaling them using an appropriate scaler before creating datapoints, rather "
+            "than using the scikit-learn `StandardScaler` (the default in Chemprop)."
+        )
+
+    def __len__(self) -> int:
+        return len(Descriptors.descList)
+
+    def __call__(self, mol: Chem.Mol) -> np.ndarray:
+        if mol.GetNumHeavyAtoms() == 0:
+            features = np.array(
+                [0 if name == "SPS" else func(mol) for name, func in Descriptors.descList], float
+            )
+        else:
+            features = np.array([func(mol) for name, func in Descriptors.descList], float)
```

Comment:
```suggestion
        features = np.array(
            [func(mol) for _, func in filter(lambda i: mol.GetNumHeavyAtoms() > 0 or i[0] != "SPS", Descriptors.descList)],
            dtype=float,
        )
```

I think this is the same and avoids duplicating a line

Reply:
Thank you for writing that with `filter`. I switched am putting `i[0] != "SPS"` first so that we don't need to call `GetNumHeavyAtoms` most of the time, only when the function name is SPS.